br/pkg/metautil/metafile.go

```diff
@@ -274,31 +274,57 @@ func (reader *MetaReader) ReadDDLs(ctx context.Context) ([]byte, error) {
 	}
 }
 
+type readSchemaConfig struct {
+	skipFiles bool
+}
+
+// ReadSchemaOption describes some extra option of reading the config.
+type ReadSchemaOption func(*readSchemaConfig)
+
+// SkipFiles is the configuration which will make the schema reader skip all files.
+// This is useful when only schema information is needed.
+func SkipFiles(conf *readSchemaConfig) {
+	conf.skipFiles = true
+}
+
+// GetBasic returns a basic copy of the backup meta.
+func (reader *MetaReader) GetBasic() backuppb.BackupMeta {
+	return *reader.backupMeta
+}
+
 // ReadSchemasFiles reads the schema and datafiles from the backupmeta.
 // This function is compatible with the old backupmeta.
-func (reader *MetaReader) ReadSchemasFiles(ctx context.Context, output chan<- *Table) error {
+func (reader *MetaReader) ReadSchemasFiles(ctx context.Context, output chan<- *Table, opts ...ReadSchemaOption) error {
 	ch := make(chan interface{}, MaxBatchSize)
 	errCh := make(chan error, 1)
 	go func() {
+		defer close(ch)
 		if err := reader.readSchemas(ctx, func(s *backuppb.Schema) { ch <- s }); err != nil {
 			errCh <- errors.Trace(err)
 		}
-		close(ch)
 	}()
 
+	cfg := readSchemaConfig{}
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
 	// It's not easy to balance memory and time costs for current structure.
 	// put all files in memory due to https://github.com/pingcap/br/issues/705
-	fileMap := make(map[int64][]*backuppb.File)
-	outputFn := func(file *backuppb.File) {
-		tableID := tablecodec.DecodeTableID(file.GetStartKey())
-		if tableID == 0 {
-			log.Panic("tableID must not equal to 0", logutil.File(file))
+	var fileMap map[int64][]*backuppb.File
+	if !cfg.skipFiles {
+		fileMap = make(map[int64][]*backuppb.File)
+		outputFn := func(file *backuppb.File) {
+			tableID := tablecodec.DecodeTableID(file.GetStartKey())
+			if tableID == 0 {
+				log.Panic("tableID must not equal to 0", logutil.File(file))
```

Comment:
make `TiDB` panic when use brie SQL :dizzy_face:

Reply:
Don't worry, TIDB will catch this panic...